demos/speech_recognition_demo/python/speech_recognition_demo.py

```diff
@@ -55,56 +65,69 @@ def get_profile(profile_name):
+
     start_time = timeit.default_timer()
     with Timer() as timer:
-        args = build_argparser().parse_args()
-        profile = get_profile(args.profile)
-
-        stt = DeepSpeechPipeline(
+        stt = DeepSpeechSeqPipeline(
             model = args.model,
             lm = args.lm,
             beam_width = args.beam_width,
             max_candidates = args.max_candidates,
             profile = profile,
             device = args.device,
             ie_extensions = [(args.device, args.cpu_extension)] if args.device == 'CPU' else [],
+            online_decoding = args.online,
         )
+    print("Loading, including network weights, IE initialization, LM, building LM vocabulary trie: {} s".format(timer.elapsed))
 
-        wave_read = wave.open(args.input, 'rb')
+    with wave.open(args.input, 'rb') as wave_read:
         channel_num, sample_width, sampling_rate, pcm_length, compression_type, _ = wave_read.getparams()
         assert sample_width == 2, "Only 16-bit WAV PCM supported"
         assert compression_type == 'NONE', "Only linear PCM WAV files supported"
         assert channel_num == 1, "Only mono WAV PCM supported"
-        audio = np.frombuffer(wave_read.readframes(pcm_length * channel_num), dtype=np.int16).reshape((pcm_length, channel_num))
-        wave_read.close()
-    print("Loading, including network weights, IE initialization, LM, building LM vocabulary trie, loading audio: {} s".format(timer.elapsed))
-    print("Audio file length: {} s".format(audio.shape[0] / sampling_rate))
-
-    # Now it is enough to call:
-    #   transcription = stt.recognize_audio(audio, sampling_rate)
-    # if you don't need to access intermediate features like character probabilities or audio features.
-
-    with Timer() as timer:
-        audio_features = stt.extract_mfcc(audio, sampling_rate=sampling_rate)
-    print("MFCC time: {} s".format(timer.elapsed))
-
-    with Timer() as timer:
-        character_probs = stt.extract_per_frame_probs(audio_features, wrap_iterator=tqdm)
-    print("RNN time: {} s".format(timer.elapsed))
+        print("Audio file length: {} s".format(pcm_length / sampling_rate))
+        print("")
+
+        audio_pos = 0
+        play_start_time = timeit.default_timer()
+        iter_wrapper = tqdm if not args.online else (lambda x: x)
+        for audio_iter in iter_wrapper(range(0, pcm_length, args.block_size)):
+            audio_block = np.frombuffer(wave_read.readframes(args.block_size * channel_num), dtype=np.int16).reshape((-1, channel_num))
+            if audio_block.shape[0] == 0:
+                break
+            audio_pos += audio_block.shape[0]
+            #
+            # It is possible to call stt.recognize_audio() for the whole audio files instead like this:
+            #   transcription1 = stt.recognize_audio(whole_audio1, sampling_rate)
+            #   transcription2 = stt.recognize_audio(whole_audio2, sampling_rate)
```

Comment:
I think it was not obvious that one `DeepSpeechSeqPipeline` object (`stt`) can be used for multiple inferences. The demo only shows inference for one file (one utterance), but in real life users will need more than one. Initialization of this object is slow, so if a user mistakenly thinks that he needs to create a new `stt` for every utterance, it will hit performance hard.